### PR TITLE
feat!: update gradle to 8.5 (#92129)

### PR DESCRIPTION
**Pull request by skiesewetter-intershop**
* update gradle to 8.5
* using JavaToolChain with Java 17
* replaced calls to deprecated code where possible
* removed scmversion-plugin
* removed detekt-plugin
* now using .github actions
* updated documentation

### CODE REVIEW

**Review by DevDavido — APPROVED** [review]

File: testdocker/Dockerfile [hunk 1]
```diff
@@ -41,7 +41,7 @@ RUN chmod a+x -R /usr/bin/dirdiff
 
 RUN addgroup --gid 150 intershop && \
     useradd -m --uid ${USERID} -g intershop --shell /bin/sh ${USERNAME} && \
-    useradd --uid 151 -g intershop --shell /bin/sh intershop
+    useradd --uid 151 -g intershop --shell /bin/sh intershop2
```
Comment (cortlepp-intershop) [inline testdocker/Dockerfile:44]: Is `intershop2` intended or a typo? If its intended, what is the reason for this change?

Comment (skiesewetter-intershop) [reply testdocker/Dockerfile:44]: This change is intended because in line 43 there's already a user added which typically is named _intershop_. I don't know how this could have worked before.

**Review by cortlepp-intershop — APPROVED** [review]
